shared-bindings/rclcpy/Node.c

```diff
@@ -0,0 +1,93 @@
+//
+// SPDX-FileCopyrightText: Copyright (c) 2025 Lucian Copeland
+//
+// SPDX-License-Identifier: MIT
+
+#include <stdint.h>
+#include "shared-bindings/rclcpy/Node.h"
+#include "shared-bindings/rclcpy/Publisher.h"
+#include "py/objproperty.h"
+#include "py/objtype.h"
+#include "py/runtime.h"
+
+static mp_obj_t rclcpy_node_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *all_args) {
+    enum { ARG_node_name, ARG_namespace };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_node_name, MP_ARG_REQUIRED | MP_ARG_OBJ },
+        { MP_QSTR_namespace, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = mp_const_none} },
+    };
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all_kw_array(n_args, n_kw, all_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+    const char *node_name = mp_obj_str_get_str(args[ARG_node_name].u_obj);
+    const char *namespace = "";
+    if (args[ARG_namespace].u_obj != mp_const_none) {
+        namespace = mp_obj_str_get_str(args[ARG_namespace].u_obj);
+    }
+
+    rclcpy_node_obj_t *self = mp_obj_malloc_with_finaliser(rclcpy_node_obj_t, &rclcpy_node_type);
+    common_hal_rclcpy_node_construct(self, node_name, context, namespace);
+    return (mp_obj_t)self;
+}
+
+static mp_obj_t rclcpy_node_create_publisher(mp_obj_t self_in, mp_obj_t topic) {
+    rclcpy_node_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    const char *topic_name = mp_obj_str_get_str(mp_obj_t);
+
+    rclcpy_publisher_obj_t *publisher = mp_obj_malloc_with_finaliser(rclcpy_publisher_obj_t, &rclcpy_publisher_type);
+    common_hal_rclcpy_publisher_construct(publisher, self,topic_name)
+    return (mp_obj_t)publisher;
+}
+static MP_DEFINE_CONST_FUN_OBJ_1(rclcpy_node_create_publisher_obj, rclcpy_node_create_publisher);
+
+static mp_obj_t rclcpy_node_get_name(mp_obj_t self_in) {
+    // TODO: probably a good idea
+    // check_for_deinit(self);
+    rclcpy_node_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    const char * name_str = common_hal_rclcpy_node_get_name(self);
+    return mp_obj_new_str(name_str);
+}
+static MP_DEFINE_CONST_FUN_OBJ_1(rclcpy_node_get_name_obj, rclcpy_node_get_name);
+
+static mp_obj_t rclcpy_node_get_namespace(mp_obj_t self_in) {
+    // TODO: probably a good idea
+    // check_for_deinit(self);
+    rclcpy_node_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    const char * namespace_str = common_hal_rclcpy_node_get_namespace(self);
+    return mp_obj_new_str(namespace_str);
+}
+static MP_DEFINE_CONST_FUN_OBJ_1(rclcpy_node_get_namespace_obj, rclcpy_node_get_namespace);
+
+static const mp_rom_map_elem_t rclcpy_node_locals_dict_table[] = {
+    { MP_ROM_QSTR(MP_QSTR_create_publisher), MP_ROM_PTR(&rclcpy_node_create_publisher_obj) },
+    { MP_ROM_QSTR(MP_QSTR_get_name), MP_ROM_PTR(&rclcpy_node_get_name_obj) },
+    { MP_ROM_QSTR(MP_QSTR_get_namespace), MP_ROM_PTR(&rclcpy_node_get_namespace_obj) },
+};
+static MP_DEFINE_CONST_DICT(rclcpy_node_locals_dict, rclcpy_node_locals_dict_table);
+
+
```

Comment:
Please add `//|` docs here.

Reply:
Thank you for the review! Sorry, haven't finished up the docs yet, was still keeping this in draft mode until I get past this memory hardfault.